kotest-extensions/src/jvmMain/kotlin/io/kotest/extensions/system/SecurityManagerExtensions.kt

```diff
@@ -62,6 +63,30 @@ class SecurityManagerTestListener(securityManager: SecurityManager?) : SecurityM
    override suspend fun afterTest(testCase: TestCase, result: TestResult) {
       resetSecurityManager()
    }
+
+   override suspend fun beforeContainer(testCase: TestCase) {
+      if (testCase.type == TestType.Container) changeSecurityManager()
+   }
+
+   override suspend fun afterContainer(testCase: TestCase, result: TestResult) {
+      if (testCase.type == TestType.Container) resetSecurityManager()
```

Comment:
We probably only need to call changeSecurityManager /  resetSecurityManager on one of the beforeAny / beforeTest cases as that will cover it for every test.